### PR TITLE
feat: Jestream configurable DiscardNew policies

### PR DESCRIPTION
**Pull request by talebzeghmi**
see **Allow Configurable Retention and Discard Policies for Jetstream EventBus** https://github.com/argoproj/argo-events/issues/3150.

Checklist:

* [x] My organization is added to [USERS.md](https://github.com/argoproj/argo-events/blob/master/USERS.md).

### CODE REVIEW

File: manifests/base/controller-manager/controller-config.yaml [hunk 1]
```diff
@@ -26,6 +26,8 @@ data:
           maxBytes: -1
           replicas: 3
           duplicates: 300s
+          retention: LimitsPolicy
```
Comment (whynowy) [inline manifests/base/controller-manager/controller-config.yaml:29]: Using Nats original config and adding comments like below?

```
            # 0: Limits, 1: Interest, 2: WorkQueue
            retention: 0
            # 0: DiscardOld, 1: DiscardNew
            discard: 0
```

Comment (talebzeghmi) [reply manifests/base/controller-manager/controller-config.yaml]: done thank you!
Is the E2E Tests (kafka) `expected sensor pod log contains 'Successfully processed trigger 'log-trigger''` failure known and expected?

File: manifests/base/controller-manager/controller-config.yaml [hunk 2]
```diff
@@ -25,7 +25,10 @@ data:
           maxAge: 168h
           maxBytes: -1
           replicas: 3
-          duplicates: 300s
```
Comment (whynowy) [inline manifests/base/controller-manager/controller-config.yaml:28]: Something wrong with the config?

Comment (talebzeghmi) [reply manifests/base/controller-manager/controller-config.yaml:28]: fixed! sorry for missing that!

File: manifests/base/controller-manager/controller-config.yaml [hunk 3]
```diff
@@ -26,6 +26,10 @@ data:
           maxBytes: -1
           replicas: 3
           duplicates: 300s
+          duplicates:           # 0: Limits, 1: Interest, 2: WorkQueue
```
Comment (whynowy) [inline manifests/base/controller-manager/controller-config.yaml:29]: Somehow it's still there... @talebzeghmi

Comment (talebzeghmi) [reply manifests/base/controller-manager/controller-config.yaml]: fixed, sorry didn't catch the typo!

**Review by whynowy — APPROVED** [review]
LGTM, thanks!

**Pull request merged**